datafusion/physical-plan/src/sorts/merge.rs

```diff
@@ -97,6 +97,23 @@ pub(crate) struct SortPreservingMergeStream<C: CursorValues> {
     /// Cursors for each input partition. `None` means the input is exhausted
     cursors: Vec<Option<Cursor<C>>>,
 
+    /// Flag indicating whether we are in the mode of round-robin
```

Comment:
Would it be possible to add some more long form documentation on the rationale / design of this polling mechanism? I think having the rationale in the code would make this easier to understand/read for future readers. 

In particular, perhaps we can adapt the (very nice) content in the `Rationale` section into the comments of `SortPreservingMergeStream`

Reply:
Added it on the flag